src/annotator/config/settings.js

```diff
@@ -3,6 +3,21 @@ import { parseJsonConfig } from '../../boot/parse-json-config';
 import configFuncSettingsFrom from './config-func-settings-from';
 import isBrowserExtension from './is-browser-extension';
 
+/**
+ * @typedef SettingsGetters
+ * @prop {string|null} annotations
+ * @prop {string} clientUrl
+ * @prop {string|null} group
+ * @prop {string} notebookAppUrl
+ * @prop {string} showHighlights
+ * @prop {string} sidebarAppUrl
+ * @prop {string|null} query
+ * @prop {(name: string, options?: Object) => (string|null)} hostPageSetting
```

Comment:
I think this type would be easier to skim if the related properties/methods were grouped together (eg. `annotations` with `query` and `group` and `notebookAppUrl` with `sidebarAppUrl`).

Writing this down makes it more obvious that this object contains a mix of generic methods (`hostPageSetting`) and a bunch of special-case getters. In future I think it would be good to try and have fewer special cases.